polus-projective-transformation-image-registration-plugin/src/parser.py

```diff
@@ -0,0 +1,117 @@
+from filepattern import FilePattern, parse_directory,get_matching,parse_filename,get_regex
+import os
+import itertools
+import numpy as np
+
+
+def parse_collection(directory_path,file_pattern,registration_variable, similarity_variable, template_image):
```

Comment:
It seems like this function could be vastly simplified. Below is some example code of how it might be done.

```python
from pathlib import Path

parser_object=FilePattern(directory_path,file_pattern)
result_dic = {}

# get variables except the registration and similarity variable
moving_variables=[var for var in variables if var not in registration_variable and var not in similarity_variable]

for f in parser_object.iterate():
    if Path(f[0]['file']).name == template_image:
        template_info = f[0]

registration_nums = {v:template_info[v] for v in parser_object.var_order if v in registration_variable}

for f in parser_object.iterate(groupby=moving_variables):
    # Skip this iteration if this contains the template image
    if template_info['file'] in [m['file'] for m in f]:
        continue
    
    # Determine the moving image
    for m in f:
        matches = True
        for v in registration_nums.keys():
            if m[v] != registration_nums[v]:
                matches = False
                break
        if matches:
            move_img = m['file']
            break

    # Store in the output dictionary
    result_dic[(template_info['file'],move_img)] = [m['file'] for m in f]